### ISSUE TITLE
pycodestyle FutureWarning is disabling AtomLinter/linter-flake8

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#243]
In GitLab by @cclauss on Jun 30, 2018, 23:23

* $ __flake8 --version__

```
/usr/local/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')

3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Darwin
```
This happens on the command line of the current production flake8 running on the current production Python on the current production Mac OSX.  It does not seem matter whether flake8 has been brew or pip installed.  The effect of this pycodestyle warning message is that AtomLinter/linter-flake8 is choking instead of eating the warning and moving on to do what it is supposed to do.

*Please describe how you installed Flake8*

Example:

```
$ pip install --user flake8
$ brew install flake8
# etc.
```

Happens either way.

**Note**: Some *nix distributions patch Flake8 arbitrarily to accommodate incompatible software versions. If you're on one of those distributions, your issue may be closed and you will be asked to open an issue with your distribution package maintainers instead.

*Please provide the exact, unmodified output of `flake8 --bug-report`*

$ __flake8 --bug-report__
```
/usr/local/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
{
  "dependencies": [
    {
      "dependency": "setuptools",
      "version": "39.2.0"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.0",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.3.1"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "1.6.0"
    }
  ],
  "version": "3.5.0"
}
```

*Please describe the problem or feature*

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

**Comment by asottile** [comment]
In GitLab by @cclauss on Jun 30, 2018, 23:24

changed the description

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jul 1, 2018, 05:39

https://github.com/PyCQA/pycodestyle/issues/728

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jul 1, 2018, 05:39

closed

**Comment by asottile** [comment]
In GitLab by @agrothberg on Jul 25, 2018, 06:47

Any chance to get a new release of `flake8` with this fix in it?

**Comment by asottile** [comment]
In GitLab by @hoylemd on Aug 14, 2018, 10:00

Can we get a status update please? This issue is quite annoying when working with python 3.7

**Comment by asottile** [comment]
In GitLab by @autoferrit on Aug 19, 2018, 22:25

I get this as well in a python `3.7.0` environment (pyenv)
```bash
$ flake8 --version
/home/autoferrit/code/src/project/.venv/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Linux

$ cat /etc/lsb-release
LSB_VERSION=1.4
DISTRIB_ID=Arch
DISTRIB_RELEASE=rolling
DISTRIB_DESCRIPTION="Arch Linux"
```

**Comment by asottile** [comment]
In GitLab by @max-wittig on Sep 4, 2018, 13:20

@sigmavirus24 Why was this issue closed?

pycodestyle needs to be updated to v2.4.0 or higher

**Comment by asottile** [comment]
In GitLab by @jason.schmedes on Sep 14, 2018, 09:41

Since they aren't going to fix this can anyone recommend a replacement?

**Comment by asottile** [comment]
In GitLab by @max-wittig on Sep 14, 2018, 09:42

I've switched everything to pycodestyle. It's good enough in itself.

**Comment by asottile** [comment]
In GitLab by @LondonAppDev on Oct 8, 2018, 24:34

Hey @sigmavirus24, appreciate the link to the GitHub issue. I understand this is fixed in the `master` branch, is there any plan to push this to PyPi?

**Comment by asottile** [comment]
In GitLab by @marcelopoars on Oct 16, 2018, 18:44

Flake8 no Python 3.6.6 está ok !

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Oct 20, 2018, 05:20

locked this issue

**Comment by asottile** [comment]
In GitLab by @kerel-fs on Mar 9, 2019, 15:34

mentioned in commit kerel-fs/python-satellitetle@968bc456cf442cc0a18846a90fdf63ef1fca68a3

**Comment by asottile** [comment]
In GitLab by @asottile on Mar 11, 2019, 10:39

Late update, but noticed this got some recent activity.  This has been fixed in flake8 3.6.0, upgrading to the latest release of flake8 should resolve this (3.7.7 is the current latest)
